src/utilities/effects.ts

```diff
@@ -4359,4 +4385,42 @@ const effects = {
   },
 };
 
+export function replacePaths(
+  modelParameters: ParametersMap | null,
+  simArgs: ArgumentsMap,
+  filenames: Record<string, string>,
+): ArgumentsMap {
+  if (modelParameters === null) {
+    return simArgs;
+  }
+  const result: ArgumentsMap = {};
+  for (const parameterName in modelParameters) {
+    const parameter: Parameter = modelParameters[parameterName];
+    const arg: Argument = simArgs[parameterName];
+    if (arg !== undefined) {
+      result[parameterName] = replacePathsHelper(parameter.schema, arg, filenames);
+    }
+  }
+  return result;
+}
+
+function replacePathsHelper(schema: ValueSchema, arg: Argument, filenames: Record<string, string>) {
+  switch (schema.type) {
+    case 'path':
+      return filenames[arg] || arg;
```

Comment:
Is this for the case when the filename was renamed to be unique? If so, maybe `filenames` isn't the most clear name for this? I'm terrible with coming up with names, but do you think something along the lines of `filenameLookup` or `filenameMap` would be more suitable? Or you could add a comment explaining the purpose of it; but with it being passed around in multiple places, it might be easier to understand if the name of it was more explicit.

Reply:
I've renamed `filenames` to a couple different things. Where they are first defined, I renamed it to `generatedFilenames`, and added a comment explaining that the gateway generates new filenames for every uploaded file.

In the `replacePaths` function, I renamed `filenames` to `pathsToReplace`, and replaced `filenames[arg] || arg` with the slightly more verbose, but hopefully more readable: https://github.com/NASA-AMMOS/aerie-ui/blob/bfce02799d0da312f958856ce636aadd94a93304/src/utilities/effects.ts#L4419-L4424